src/lib.rs

```diff
@@ -215,6 +215,39 @@ impl<T: Hash, S: BuildHasher> HashRing<T, S> {
     }
 }
 
+pub struct HashRingIterator<T> {
+    ring: std::vec::IntoIter<T>,
+    index: usize,
+}
+
+impl<T> Iterator for HashRingIterator<T> {
+    type Item = T;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let out = self.ring.next();
+        self.index += 1;
+        out
+    }
+}
+
+impl<T: Clone> IntoIterator for HashRing<T> {
+    type Item = T;
+
+    type IntoIter = HashRingIterator<T>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        HashRingIterator {
+            ring: self
+                .ring
+                .iter()
+                .map(|n| n.node.clone())
+                .collect::<Vec<T>>()
```

Comment:
To avoid needing to collect the items from the vector into an iterator here, what do you think about using the internal iterator from the vector directly and mapping `Node<T>` to `T` in the hash ring iterator's `next` method? For example:

```rust
pub struct HashRingIterator<T> {
    iter: std::vec::IntoIter<Node<T>>,
}

impl<T> Iterator for HashRingIterator<T> {
    type Item = T;

    fn next(&mut self) -> Option<Self::Item> {
        self.iter.next().map(|n| n.node)
    }
}
```

Reply:
I didn't want to do it this way as it would expose the internal `Node` type and expose a layer of abstraction. This is much cleaner way of doing it. Changes coming soon